rasa_nlu/data_router.py

```diff
@@ -234,3 +257,63 @@ def training_errback(failure):
         result.addErrback(training_errback)
 
         return result
+
+    def evaluate(self, data, parameters):
+        # type: (Text, Dict[Text, Any]) -> Dict[Text, Any]
+        """Perform a model evaluation."""
+
+        if PY3:
+            f = tempfile.NamedTemporaryFile("w+", suffix="_training_data",
+                                            delete=False, encoding="utf-8")
+            f.write(data)
+        else:
+            f = tempfile.NamedTemporaryFile("w+", suffix="_training_data",
+                                            delete=False)
+            f.write(data.encode("utf-8"))
+        f.close()
+        test_data = load_data(f.name)
+
+        project = parameters.get(
+            "project") or RasaNLUConfig.DEFAULT_PROJECT_NAME
+        model = parameters.get("model")
+
+        if not (project and project in self.project_store):
+            raise InvalidProjectError("Missing project name to evaluate")
+
+        preds_json = []
```

Comment:
i think this loop can go in a separate function as well.

logging should most likely not be on `logger.info` but `debug`